src/Tribe/Assets.php

```diff
@@ -0,0 +1,286 @@
+	}
+
+	/**
+	 * Register the Methods in the correct places
+	 */
+	private function __construct() {
+		// Hook the actual rendering of notices
+		add_action( 'wp_enqueue_scripts', array( $this, 'hook' ), 1 );
+		add_action( 'admin_enqueue_scripts', array( $this, 'hook' ), 1 );
+		add_action( 'login_enqueue_scripts', array( $this, 'hook' ), 1 );
+	}
+
+	public function hook() {
+		foreach ( $this->assets as $asset ) {
+			if ( 'js' === $asset->type ) {
+				wp_register_script( $asset->slug, $asset->url, $asset->deps, $asset->version, $asset->in_footer );
+			} else {
+				wp_register_style( $asset->slug, $asset->url, $asset->deps, $asset->version, $asset->media );
+			}
+
+			if ( ! is_string( $asset->action ) ) {
+				continue;
+			}
+
+			add_action( $asset->action, array( $this, 'enqueue' ), $asset->priority );
+		}
+	}
+
+	public function enqueue() {
+		foreach ( $this->assets as $asset ) {
+			// Skip if we are not on the correct filter
+			if ( current_filter() !== $asset->action ) {
+				continue;
+			}
+
+			// If here is no action we are just registering this asset
+			if ( ! is_string( $asset->action ) ) {
+				continue;
+			}
+
+			// If we have a set of conditionals we loop on then and get if they are true
+			if ( ! empty( $asset->conditionals ) ) {
+				$conditionals = array();
+				foreach ( $asset->conditionals as $conditional ) {
+					$conditionals[] = call_user_func_array( $conditional, array() );
+				}
+
+				// If any of them are false we skip this one
+				if ( in_array( false, $conditionals ) ) {
+					continue;
+				}
+			}
+
+			if ( 'js' === $asset->type ) {
+				wp_enqueue_script( $asset->slug );
+
+				// Only localize on JS and if we have data
+				if ( ! empty( $asset->localize ) ) {
+					wp_localize_script( $asset->slug, $asset->localize->name, $asset->localize->data );
+				}
+			} else {
+				wp_enqueue_style( $asset->slug );
+			}
+		}
+	}
+
+	/**
+	 * Returns the path to a minified version of a js or css file, if it exists.
+	 * If the file does not exist, returns false.
+	 *
+	 * @param string $url   The path or URL to the un-minified file.
+	 *
+	 * @return string|false The path/url to minified version or false, if file not found.
+	 */
+	public static function maybe_get_min_file( $url ) {
+		if ( ! defined( 'SCRIPT_DEBUG' ) || SCRIPT_DEBUG === false ) {
+			if ( substr( $url, - 3, 3 ) === '.js' ) {
+				$url = substr_replace( $url, '.min', - 3, 0 );
+			}
+
+			if ( substr( $url, - 4, 4 ) === '.css' ) {
+				$url = substr_replace( $url, '.min', - 4, 0 );
+			}
+		}
+
+		$file = str_replace( WP_CONTENT_URL, WP_CONTENT_DIR, $url );
+		if ( file_exists( $file ) ) {
+			return $url;
+		} else {
+			return false;
+		}
+	}
+
+	/**
+	 * Stores all the Assets and it's configurations
+	 *
+	 * @var array
+	 */
+	private $assets = array();
+
+	/**
+	 * Register a Asset and attach a callback to the required action to display it correctly
+	 *
+	 * @param  string   $slug      Slug to save the asset
```

Comment:
And `$file`, `$deps`, `$action`. Extra `$callback`. Now that I think about it, I think this is just the wrong docblock :)